### ISSUE TITLE
Feature request: matrix to column vector method.

### ISSUE THREAD

**Issue by GJBoth** [numpy/numpy#14672]
Neural networks take in vector data with shape (N x 1), i.e. a column vector, but the data I'm working with is usually generated on a grid, i.e. (A x B). Thus, to feed my data to the network, I have to .flatten() and add an axis or use .reshape(matrix.size, 1). This is an operation I use very often and I suspect many more with me, and thus I think a method or option performing this operation would be of value. I think this could be implemented in two ways:

Option 1) ndarray.vector() method:
Add a .vector() method which flattens an array but keeps the outer axis, i.e. (N x M) -> (N*M, 1)

Option 2) .flatten(maintain_axis=True)
Add a maintain_axis keyword so that the flattened axis is maintained.

What are your thoughts?

**Comment by mattip** [comment]
`arr.reshape(arr.size, 1)` seems pretty easy and should be performant. What am I missing?

**Comment by GJBoth** [comment]
I has nothing to do with performance; it would purely be syntactic sugar. I feel being able to write .vector() would be a lot nicer than arr.reshape(arr.size, 1), especially since it's used quite often when working with neural nets.

**Comment by eric-wieser** [comment]
> Neural networks take in vector data with shape (N x 1), i.e. a column vector

This strikes me as the matlab way to phrase the problem. Neural nets take a vector of shape `(N,)`, not a column or row vector. Are you sure you need to insert this extra axis in the first place?

**Comment by GJBoth** [comment]
@eric-wieser   Pytorch and Tensorflow require you be explicit about the axes, and since I'm often reshaping being explicit about the axes is very useful. Furthemore, if you have multiple inputs your input is a matrix with the features on the second axis and then, for the sake of consistency, feeding a Nx1 matrix is better, imo.

**Comment by pmli** [comment]
What about `.reshape(-1, 1)`?
